altsrc/map_input_source.go

```diff
@@ -16,6 +16,11 @@ type MapInputSource struct {
 	valueMap map[interface{}]interface{}
 }
 
+//NewMapInputSource create a new MapInputSource type
```

Comment:
I'm not super familiar with the `altsrc` package, so it took me a minute to figure out what this was used for. Quick suggestion on a slightly more detailed Godoc:

```suggestion
// NewMapInputSource creates a new MapInputSource for implementing custom input sources.
```